superset/assets/src/visualizations/deckgl/utils.js

```diff
@@ -0,0 +1,88 @@
+import d3 from 'd3';
+import getSequentialSchemeRegistry from '../../modules/colors/SequentialSchemeRegistrySingleton';
+import { colorScalerFactory } from '../../modules/colors';
+
+export function getBreakPoints(fd, features) {
+  if (fd.break_points === undefined || fd.break_points.length === 0) {
+    // compute evenly distributed break points based on number of buckets
+    const numBuckets = fd.num_buckets
+      ? parseInt(fd.num_buckets, 10)
+      : 10;
+    const [minValue, maxValue] = d3.extent(features, d => d[fd.metric]);
+    const delta = (maxValue - minValue) / numBuckets;
+    const precision = delta === 0
+      ? 0
+      : Math.max(0, Math.ceil(Math.log10(1 / delta)));
+    return Array(numBuckets + 1)
+      .fill()
+      .map((_, i) => (minValue + i * delta).toFixed(precision));
+  }
+  return fd.break_points.sort((a, b) => parseFloat(a) - parseFloat(b));
+}
+
+export function hexToRGB(hex, alpha = 255) {
+  if (!hex) {
+    return [0, 0, 0, alpha];
+  }
+  const r = parseInt(hex.slice(1, 3), 16);
+  const g = parseInt(hex.slice(3, 5), 16);
+  const b = parseInt(hex.slice(5, 7), 16);
+  return [r, g, b, alpha];
+}
+
+export function getBreakPointColorScaler(fd, features) {
+  const breakPoints = fd.break_points || fd.num_buckets
+    ? getBreakPoints(fd, features)
+    : null;
+  const colors = Array.isArray(fd.linear_color_scheme)
+    ? fd.linear_color_scheme
+    : getSequentialSchemeRegistry().get(fd.linear_color_scheme).colors;
+
+  let scaler;
+  let maskPoint;
+  if (breakPoints !== null) {
+    // bucket colors into discrete colors
```

Comment:
There is `d3.scale.quantize()`

Reply:
I looked at it and it didn't work in this case because we want to quantize into buckets of different widths (eg, `0-10` and `10-100`).

Reply:
got it.